torch/csrc/jit/tensorexpr/schedule.cpp

```diff
@@ -399,6 +399,126 @@ class FunctionInliner : public IRMutator {
   std::unordered_set<const Var*> func_var_set_;
 };
 
+/**
+ * Actual algorithm for "inlining" of rand().  Bind variables to rand() calls.
+ */
+class RandomInliner : public IRMutator {
+ public:
+  RandomInliner(const std::unordered_set<Function*>& funcs) : funcs_(funcs) {
+    for (auto const* func : funcs_) {
+      func_vars_.insert(func->func_var(0));
+    }
+  }
+
+  static bool is_inner_loop(const For* loop) {
+    Block* body = dynamic_cast<Block*>(loop->body());
+    if (!body) {
```

Comment:
This should be turned to an assert/removed. Bodies of For loops are blocks by construction.